### ISSUE TITLE
how to get the train/val split for cub dataset

### ISSUE THREAD

**Issue by Adasunnylily** [Trustworthy-ML-Lab/Label-free-CBM#2]
hi, great work!
Could you please tell me how to get the train/val split cub dataset?
Thanks a lot!

**Comment by tuomaso** [comment]
Thanks! The train/val split for CUB can be created using the `data/split_cub_dataset.py` file. This should also be automatically done if you download CUB using the `download_cub.sh` script we have provided.

**Comment by Adasunnylily** [comment]
thanks for the prompt!
I split the cub dataset and run as set in the code,however I cant get the same ACC reported in table2, could you please tell me any mistake I have made?  The code are as follows:
root@b1e327811162:/home/zhangyx3/Label-free-CBM# python train_cbm.py --con
cept_set /home/zhangyx3/Label-free-CBM/data/concept_sets/cub_filtered.txt --print --dataset cub
100%|████████████████████████████████████████████| 12/12 [00:31<00:00,  2.61s/it]
100%|████████████████████████████████████████████| 12/12 [00:17<00:00,  1.47s/it]
100%|████████████████████████████████████████████| 12/12 [00:19<00:00,  1.62s/it]
100%|████████████████████████████████████████████| 12/12 [00:19<00:00,  1.66s/it]
Deleting a Mohs hardness of 10, CLIP top5:0.245
Deleting a built-in keyboard, CLIP top5:0.217
Deleting a hard, crystalline structure, CLIP top5:0.240
Deleting a loud cawing sound, CLIP top5:0.246
Deleting a loud, harsh cry, CLIP top5:0.236
Deleting a red ring around the eye, CLIP top5:0.249
Deleting all black body, CLIP top5:0.242
Deleting hairy body, CLIP top5:0.227
Deleting orange eyes, CLIP top5:0.250
Deleting pale belly, CLIP top5:0.244
Deleting pattern, CLIP top5:0.226
Deleting red skin around the face, CLIP top5:0.250
Deleting shaggy appearance, CLIP top5:0.239
Deleting shape, CLIP top5:0.245
Deleting white sides, CLIP top5:0.241
Step:0, Avg train similarity:-0.0021, Avg val similarity:0.1020
Best step:100, Avg val similarity:0.6260
Deleting a high dispersion, Iterpretability:0.418
Deleting a raucous call, Iterpretability:0.444
Deleting a streaked chest, Iterpretability:0.400
Deleting a white stripe on the tail, Iterpretability:0.421
Deleting long tail feathers, Iterpretability:0.449
Deleting loud, rasping voice, Iterpretability:0.448
100%|████████████████████████████████████████| 1000/1000 [05:20<00:00,  3.12it/s]
did not converge at 1000 iterations (criteria False)
(0) lambda 0.0007, loss 1.9931, acc 0.8361 [val acc 0.6287] [test acc -1.0000], sparsity 0.09840974212034384 [6869/69800], time 321.09886717796326, lr 0.1000

**Comment by tuomaso** [comment]
To train on cub you need to change some hyperparameters, the exact commands are found in` training_commands.txt`.  I.e.
`python train_cbm.py --dataset cub --backbone resnet18_cub --concept_set data/concept_sets/cub_filtered.txt --feature_layer features.final_pool --clip_cutoff 0.26 --n_iters 5000 --lam 0.0002`

**Comment by Adasunnylily** [comment]
Thanks a lot! I have tried the provided code for cub and got similar results. I still have another problem on ImageNet impletation.

I use the command `python train_cbm.py --dataset imagenet --backbone resnet50 --concept_set data/concept_sets/imagenet_filtered.txt --clip_cutoff 0.28 --n_iters 80 --lam 0.0001`  as in training command.txt and I got errors as follow:
Traceback (most recent call last):
  File "train_cbm.py", line 256, in <module>
    train_cbm_and_save(args)
  File "train_cbm.py", line 61, in train_cbm_and_save
    utils.save_activations(clip_name = args.clip_name, target_name = args.backbone, 
  File "/home/zhangyx3/Label-free-CBM/utils.py", line 98, in save_activations
    target_model, target_preprocess = data_utils.get_target_model(target_name, device)
  File "/home/zhangyx3/Label-free-CBM/data_utils.py", line 107, in get_target_model
    weights = eval("models.{}_Weights.IMAGENET1K_V1".format(target_name_cap))
  File "<string>", line 1, in <module>
AttributeError: module 'torchvision.models' has no attribute 'ResNet50_Weights'

Many thanks for your generous help.
